src/worker/tasks/updateMinedTx.ts

```diff
@@ -48,28 +48,57 @@ export const updateMinedTx = async () => {
-                    queueId: tx.id,
-                    status: TransactionStatusEnum.Cancelled,
-                  });
-
-                  reportUsageForQueueIds.push({
-                    input: {
-                      fromAddress: tx.fromAddress || undefined,
-                      toAddress: tx.toAddress || undefined,
-                      value: tx.value || undefined,
-                      chainId: tx.chainId || undefined,
-                      transactionHash: tx.transactionHash || undefined,
-                      provider: provider.connection.url || undefined,
-                      msSinceSend: Date.now() - tx.sentAt!.getTime(),
-                    },
-                    action: UsageEventTxActionEnum.CancelTx,
-                  });
+                  try {
+                    await cancelTransactionAndUpdate({
+                      queueId: tx.id,
+                      pgtx,
+                    });
+
+                    sendWebhookForQueueIds.push({
+                      queueId: tx.id,
+                      status: TransactionStatusEnum.Cancelled,
+                    });
+
+                    reportUsageForQueueIds.push({
+                      input: {
+                        fromAddress: tx.fromAddress || undefined,
+                        toAddress: tx.toAddress || undefined,
+                        value: tx.value || undefined,
+                        chainId: tx.chainId || undefined,
+                        transactionHash: tx.transactionHash || undefined,
+                        provider: provider.connection.url || undefined,
+                        msSinceSend: Date.now() - tx.sentAt!.getTime(),
+                      },
+                      action: UsageEventTxActionEnum.CancelTx,
+                    });
+                  } catch (error) {
+                    await updateTx({
+                      pgtx,
+                      queueId: tx.id,
+                      data: {
+                        status: TransactionStatusEnum.Errored,
+                        errorMessage: "Transaction timed out.",
+                      },
+                    });
+
+                    sendWebhookForQueueIds.push({
+                      queueId: tx.id,
+                      status: TransactionStatusEnum.Errored,
+                    });
+
+                    reportUsageForQueueIds.push({
+                      input: {
+                        fromAddress: tx.fromAddress || undefined,
+                        toAddress: tx.toAddress || undefined,
+                        value: tx.value || undefined,
+                        chainId: tx.chainId || undefined,
+                        transactionHash: tx.transactionHash || undefined,
+                        provider: provider.connection.url || undefined,
+                        msSinceSend: Date.now() - tx.sentAt!.getTime(),
+                      },
+                      action: UsageEventTxActionEnum.NotSendTx,
```

Comment:
We should report these as "errors" (Engine problem to fix) instead of "not send" (customer problem to fix)

Reply:
we do report `notsendTx` in other places too, like the retryTx Flow also i believe should follow same notion. I will update